buffer/src/main/java/io/netty/buffer/AbstractByteBuf.java

```diff
@@ -332,12 +332,12 @@ public int ensureWritable(int minWritableBytes, boolean force) {
 
     @Override
     public ByteBuf order(ByteOrder endianness) {
-        if (endianness == null) {
-            throw new NullPointerException("endianness");
-        }
         if (endianness == order()) {
             return this;
         }
+        if (endianness == null) {
```

Comment:
this change looks unnecessary, can you revert it?

Reply:
This change eliminates a null check in the most common case (BE) and doesn't change the number of checks in the other cases, is there a reason the other way round is preferable?

Reply:
I don't have a preference here... that said null checks are really cheap. So whatever is more readable is the preferred way imho :)

Reply:
I don't love that it changes the git blame, but that seems reasonable.  shipit.